### PR TITLE
Bump kapp to 0.39.0 

### LINKED ISSUE #347: Bump kapp to v0.38.0
kapp-controller should upgrade to the [latest version of kapp](https://github.com/vmware-tanzu/carvel-kapp/releases/tag/v0.38.0). As part of doing so, we should also update e2e tests for kapp-controller to take advantage of kapp's ability to wait for reconciliation of PackageInstalls/PackageRepositories as described in #305.

### PR DESCRIPTION
**Pull request by danielhelfand**
Closes #347 

Bumps kapp to 0.39.0 and updates associated e2e tests.

### CODE REVIEW

File: test/e2e/kappcontroller/app_secret_configmap_reconcile_test.go [hunk 1]
```diff
@@ -209,3 +209,18 @@ data:
 		})
 	})
 }
+
+func retry(t *testing.T, timeout time.Duration, f func() error) {
+	var err error
+	stopTime := time.Now().Add(timeout)
```
Comment (joe-kimmel-vmw) [inline test/e2e/kappcontroller/app_secret_configmap_reconcile_test.go:215]: fwiw i'd vote for counting retry iterations over measuring time -  there's several cases where time can move around on you,  for instance if we're running on a virtual host which suspends and resumes,  or if the host clock has skewed and during our test run it resyncs with NTP... ( see also,  https://gist.github.com/timvisee/fcda9bbdff88d45cc9061606b4b923ca )

Comment (danielhelfand) [reply test/e2e/kappcontroller/app_secret_configmap_reconcile_test.go:215]: I think that's fair. Can we open a separate issue for this to keep this pr focused on kapp 0.38.0?

Comment (joe-kimmel-vmw) [reply test/e2e/kappcontroller/app_secret_configmap_reconcile_test.go:215]: cool: https://github.com/vmware-tanzu/carvel-kapp-controller/issues/351

**Comment by cppforlife** [comment]
lgtm

**Pull request merged**